zombies.js

```diff
@@ -78,136 +90,252 @@
+   *
+   * If the item is not in the pack, return a message with the item name saying
+   *   nothing was discarded since the item could not be found.
+   * Return false in this case.
+   *
+   * You should be able to invoke this function on a Player instance.
+   *
+   * @name discardItem
+   * @param {Item/Weapon/Food} item   The item to discard.
+   * @return {boolean} true/false     Whether player was able to remove item from pack.
+   */
+  discardItem(item) {
+    let itemIndex = this.isItemInPack(item);
+    if (itemIndex === false) {
+      console.log(`${item.name} was not found. Nothing was discarded`);
+      return false;
+    } else {
+      console.log(`Player: ${this._name}, Item: ${item.name} \nItem was discarded.`)
+      this.getPack().splice(itemIndex, 1);
+      return true;
+    }
+  }
+
+  /**
+   * Player Class Method => equip(itemToEquip)
+   * -----------------------------
+   * Player equips a weapon item.
+   *
+   * Player can only equip Weapon instances.
+   * Player can only equip weapon items from their pack.
+   *
+   * If the player already has a weapon equipped (the equipped property
+   *   is set to an Item), find the itemToEquip in the pack and replace
+   *   it with the currently equipped item.  Then set the equipped property
+   *   to the itemToEquip.
+   * However, if the player doesn't already have a weapon equipped, simply
+   *   equip that item and remove it from the pack.
+   * You should be able to invoke this function on a Player instance.
+   *
+   * @name equip
+   * @param {Weapon} itemToEquip  The weapon item to equip.
+   */
+  equip(itemToEquip) {
+    let itemIndex = this.isItemInPack(itemToEquip);
+    if (itemToEquip instanceof (Weapon) && itemIndex !== false) {
+      if (this._equipped) {
+        this.getPack()[itemIndex] = this._equipped;
+      } else {
+        this.getPack().splice(itemIndex, 1);
+      }
+      this._equipped = itemToEquip;
+    }
+  }
+
+  /**
+   * Player Class Method => eat(itemToEat)
+   * -----------------------------
+   * Player eats a food item, restoring their health.
+   *
+   * Player can only eat Food instances.
+   * Player can only eat food items from their pack.
+   *
+   * Remove itemToEat from the pack.
+   * Increase the player's health by the food's energy amount, but do not
+   *   exceed the player's max health.  If exceeded, simply set player's health
+   *   to max health instead.
+   * To access the player's max health, be sure to use Player's getMaxHealth method.
+   * You should be able to invoke this function on a Player instance.
+   *
+   * @name eat
+   * @param {Food} itemToEat  The food item to eat.
+   */
+  eat(itemToEat) {
+    let itemIndex = this.isItemInPack(itemToEat);
+    if (itemToEat instanceof (Food) && itemIndex !== false) {
+      this.getPack().splice(itemToEat, 1);
+      this._health = this._health + itemToEat.energy > this.getMaxHealth() ? this.getMaxHealth() : this._health + itemToEat.energy;
+    }
```

Comment:
```js
this.health += itemToEat.energy;
if(this.health > this.getMaxHealth()) {
  this.health = this.getMaxHealth();
}
```
I think the ternary is a bit hard to follow is this case.
You have a `.health` setter, so you should use it.